staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go

```diff
@@ -39,6 +39,7 @@ type GarbageCollectionPolicy string
 const (
 	DeleteDependents GarbageCollectionPolicy = "DeleteDependents"
 	OrphanDependents GarbageCollectionPolicy = "OrphanDependents"
+	Unsupported      GarbageCollectionPolicy = "Unsupported"
```

Comment:
> Also, it seems if someone deletes an "unsupported" resource with either "Orphan" or "foreground deletion", the object will be deleted immediately.

I don't think so.  We enumerate the types here: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/graph_builder.go#L235-L253, so they are never observed in the graph as being in or out.  They are just missing and never acted upon.